airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java

```diff
@@ -130,6 +143,67 @@ public static Map<AirbyteStreamNameNamespacePair, TableSizeInfo> getTableSizeInf
     return tableSizeInfoMap;
   }
 
+  /**
+   * Iterates through each stream and find the max cursor value and the record count which has that
+   * value based on each cursor field provided by the customer per stream This information is saved in
+   * a Hashmap with the mapping being the AirbyteStreamNameNamespacepair -> CursorBasedStatus
+   *
+   * @param database the source db
+   * @param streams streams to be synced
+   * @param stateManager stream stateManager
+   * @return Map of streams to statuses
+   */
+  public static Map<io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair, CursorBasedStatus> getCursorBasedSyncStatusForStreams(final JdbcDatabase database,
+                                                                                                                                        final List<ConfiguredAirbyteStream> streams,
+                                                                                                                                        final MySqlCursorBasedStateManager stateManager,
+                                                                                                                                        final String quoteString) {
+
+    final Map<io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair, CursorBasedStatus> cursorBasedStatusMap = new HashMap<>();
+    streams.forEach(stream -> {
+      try {
+        final String name = stream.getStream().getName();
+        final String namespace = stream.getStream().getNamespace();
+        final String fullTableName =
+            getFullyQualifiedTableNameWithQuoting(namespace, name, quoteString);
+
+        final Optional<CursorInfo> cursorInfoOptional =
+            stateManager.getCursorInfo(new io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair(name, namespace));
+        if (cursorInfoOptional.isEmpty()) {
+          throw new RuntimeException(String.format("Stream %s was not provided with an appropriate cursor", stream.getStream().getName()));
+        }
+
+        LOGGER.info("Querying max cursor value for {}.{}", namespace, name);
+        final String cursorField = cursorInfoOptional.get().getCursorField();
+        final String cursorBasedSyncStatusQuery = String.format(MAX_CURSOR_VALUE_QUERY,
+            cursorField,
```

Comment:
We should escape quote cursor fields like in this PR : https://github.com/airbytehq/airbyte/pull/30059/files